src/test/scala/com/cognite/sdk/scala/common/RetryWhile.scala

```diff
@@ -8,29 +8,26 @@ import org.scalatest.exceptions.TestFailedException
 
 trait RetryWhile {
   def retryWithExpectedResult[A](
-        action: => A,
-        result: Option[A],
-        shouldRetry: Seq[A => Assertion],
-        retriesRemaining: Int = Constants.DefaultMaxRetries,
-        initialDelay: FiniteDuration = Constants.DefaultInitialRetryDelay
-      ): A = {
+    action: => A,
+    assertion: A => Assertion,
+    retriesRemaining: Int = Constants.DefaultMaxRetries,
+    initialDelay: FiniteDuration = Constants.DefaultInitialRetryDelay
+  ): A = {
     val exponentialDelay = (Constants.DefaultMaxBackoffDelay / 2).min(initialDelay * 2)
     val randomDelayScale = (Constants.DefaultMaxBackoffDelay / 2).min(Constants.DefaultInitialRetryDelay * 2).toMillis
     val nextDelay = Random.nextInt(randomDelayScale.toInt).millis + exponentialDelay
-    shouldRetry.foreach(assertion => {
-        try {
-          assertion(result.getOrElse(action))
-        } catch {
-          case testFailed: TestFailedException =>
-          if (retriesRemaining > 0) {
-            Thread.sleep(initialDelay.toMillis)
-            val actionValue = action
-            retryWithExpectedResult[A](action, Some(actionValue), Seq(assertion), retriesRemaining - 1, nextDelay)
-          } else {
-            throw testFailed
-          }
-        }
-      })
-    result.getOrElse(action)
+    try {
+      val result = action
```

Comment:
Could you try making this PR into a minimal change that just fixes the usage of `action` in `retryWithExpectedResults`? I don't think we should pass the result as an argument, I agree with removing that, but doing something like this instead, perhaps?

```scala
val actionResult = action
try {
  shouldRetry.foreach { assertion =>
    assertion(actionResult)
  }
} catch {
  case testFailed: TestFailedException if (retriesRemaining > 0) =>
    Thread.sleep(initialDelay.toMillis)
    retryWithExpectedResult[A](action, shouldRetry, retriesRemaining - 1, nextDelay)
}
actionResult
```

Reply:
1. Should the `action` invocation be outside of the `try` block or inside? I can put it, just wasn't sure what's the right thing (it was mixed before...)
2. I removed the option to pass in multiple asserts as this was triggering multiple requests (and was useless). I don't really see why would we want to keep the API, but change the behavior so much. Do think multiple asserts may actually be useful?

Reply:
Ah, right. So. Inside, and it should be
```
try {
  val actionResult = action
  shouldRetry.foreach(assertion => assertion(actionResult))
  actionResult
} catch {
  case ...
    Thread.sleep(...)
    retryWithExpectedResult[A](...)
}
```

The triggering of multiple requests should be fixed by evaluating `action` once and only once per call to `retryWithExpectedResult` (up to the initial `retriesRemaining` times).

Reply:
Yes, I think you got it right, actually. Nice :)
Can still move the `if (retriesRemaining > 0)` to the `case` line, and the indentation is (and was) off in the case statement. In addition, we don't need to do "else" if we make that change.